hashing/hashing.go

```diff
@@ -196,21 +234,51 @@ func (t *CarbonHashRing) String() string {
 		len(t.nodes), t.replicas, len(t.ring), strings.Join(servers, " "))
 }
 
+func (t *FNV1aHashRing) String() string {
+	servers := make([]string, 0)
+
+	for i := 0; i < len(t.nodes); i++ {
+		servers = append(servers, t.nodes[i].String())
+	}
+	return fmt.Sprintf("[fnv1a: %d nodes, %d replicas, %d ring members %s]",
+		len(t.nodes), t.replicas, len(t.ring), strings.Join(servers, " "))
+}
+
 func (t *CarbonHashRing) Replicas() int {
 	return t.replicas
 }
 
+func (t *FNV1aHashRing) Replicas() int {
+	return t.replicas
+}
+
 func (t *CarbonHashRing) SetReplicas(r int) {
 	t.replicas = r
 }
 
+func (t *FNV1aHashRing) SetReplicas(r int) {
+	t.replicas = r
+}
+
 func (t *CarbonHashRing) AddNode(node Node) {
-	//log.Printf("insertRing(): %s", node.KeyValue())
+	//log.Printf("insertRing(): %s", node.CarbonKeyValue())
+	t.nodes = append(t.nodes, node)
+	for i := 0; i < t.replicas; i++ {
+		var e RingEntry
+		replica_key := fmt.Sprintf("%s:%d", node.CarbonKeyValue(), i)
+		e.position = computeCarbonRingPosition(replica_key)
+		e.node = node
+		t.ring = insertRing(t.ring, e)
+	}
+}
+
+func (t *FNV1aHashRing) AddNode(node Node) {
+	//log.Printf("insertRing(): %s", node.FNV1aKeyValue())
 	t.nodes = append(t.nodes, node)
 	for i := 0; i < t.replicas; i++ {
 		var e RingEntry
-		replica_key := fmt.Sprintf("%s:%d", node.KeyValue(), i)
-		e.position = computeRingPosition(replica_key)
+		replica_key := fmt.Sprintf("%d-%s", node.FNV1aKeyValue(), i)
```

Comment:
Something is backward here.  It looks like the format string should be `"%s-%d"`